builtin/providers/template/resource.go

```diff
@@ -24,13 +23,22 @@ func resource() *schema.Resource {
 		Read:   Read,
 
 		Schema: map[string]*schema.Schema{
+			"template": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Contents of the template",
+				ForceNew:    true,
+			},
 			"filename": &schema.Schema{
 				Type:        schema.TypeString,
-				Required:    true,
+				Optional:    true,
```

Comment:
worth adding `ConflictsWith` as well? Or warn that `filename` is ignored if `template` is provided? Minor thought